client/web/compose/src/components/ModuleFields/Editor/File.vue

```diff
@@ -127,6 +212,92 @@ export default {
         this.value = attachmentID
       }
     },
+    async initializeWebcam () {
+      await this.$nextTick()
+
+      this.video = this.$refs.video
+      this.canvas = this.$refs.canvas
+      this.captureButton = this.$refs.captureButton
+      this.uploadButton = this.$refs.uploadButton
+      this.closeButton = this.$refs.closeButton
+
+      this.startWebcam()
+    },
+
+    startWebcam () {
+      // Get access to the camera
+      navigator.mediaDevices.getUserMedia({ video: true })
+        .then(stream => {
+          this.stream = stream
+          this.video.srcObject = stream
+          this.video.style.display = 'block'
+          this.closeButton.style.display = 'inline-block'
+
+          if (this.hasCapturedImage) {
+            this.uploadButton.style.display = 'inline-block'
+          } else {
+            this.captureButton.style.display = 'inline-block'
+          }
+        })
+        .catch(err => {
+          console.error('Error accessing the camera: ' + err)
+        })
+    },
+
+    openCamera () {
+      this.$refs.modal.show()
+    },
+    capturePhoto () {
+      const video = this.$refs.video
+      const canvas = document.createElement('canvas')
+      canvas.width = video.videoWidth
+      canvas.height = video.videoHeight
+      canvas.getContext('2d').drawImage(video, 0, 0, canvas.width, canvas.height)
+
+      this.capturedImage = canvas.toDataURL('image/jpeg')
+      this.hasCapturedImage = true
+
+      this.stopWebcam()
+    },
+
+    uploadCapturedImage () {
+      if (this.capturedImage) {
+        fetch(this.capturedImage)
+          .then(res => res.blob())
+          .then(blob => {
+            const file = new File([blob], 'webcam-image.jpg', { type: 'image/jpeg' })
+
+            const uploader = this.$refs.uploader
+            uploader.$refs.dropzone.addFile(file)
+          })
+
+        this.$refs.modal.hide()
+      }
+    },
+
+    stopWebcam () {
+      if (this.stream) {
+        this.stream.getTracks().forEach(track => track.stop())
+      }
+    },
+    closeCamera () {
+      if (this.hasCapturedImage) {
+        this.capturedImage = null
+        this.hasCapturedImage = false
+
+        this.startWebcam()
+      } else {
+        this.stopWebcam()
+        this.$refs.modal.hide()
+      }
+    },
   },
 }
 </script>
+
+<style>
+.video-modal-width .modal-dialog {
```

Comment:
Make sure the modal body is as big as the canvas (no padding should be there).
Height should be constant so that the spinner will be centered as well.